### ISSUE TITLE
Misleading error message when using test template without arguments

### ISSUE THREAD

**Issue by HenrikSchuette** [robotframework/robotframework#4459]
In RF version 5.0.1 (Python 3.7.0) the following code returns the error message "Test contains no keywords."

Expected message: "Keyword 'My Template' expected 2 arguments, got 0."

```
*** Settings ***
Test Template    My Template


*** Test Cases ***
Test with wrong message        


*** Keywords ***
My Template
    [Arguments]    ${arg1}    ${arg2}
    Log    ${arg1}
    Log    ${arg2}
```

**Comment by pekkaklarck** [comment]
I agree the error is a bit misleading. The keyword isn't called so the error you propose wouldn't work well either, though. I think it could be changed to this:
> Test cannot be empty.

**Comment by HenrikSchuette** [comment]
I don't quite understand why the keyword is not called. For me the error message should be analog to what happens when I call the template with three arguments:

```
*** Settings ***
Test Template    My Template

*** Test Cases ***
Test with wrong message    arg1    arg2    arg3

*** Keywords ***
My Template
    [Arguments]    ${arg1}    ${arg2}
    Log    ${arg1}
    Log    ${arg2}
```

Message: Keyword 'My Template' expected 2 arguments, got 3.

**Comment by pekkaklarck** [comment]
If the keyword would be called, it would be ok to have an empty test if your template doesn't have any mandatory parameters. That's not what we want, we want it to be an error if a test (or a keyword) is empty.

**Comment by Snooz82** [comment]
But Pekka, i think that is the problem.
Why is it not possible to take a Template without arguments.

You can design it, it can be parsed, but the user just get a strange information.
Information like `Templated test cases need arguments` or so would be giving a hint.

**Comment by Snooz82** [comment]
[Runnable Example](https://robotframework.org/code/?codeProject=N4IgdghgtgpiBcIDCBXAzgFwPZQAQGMsATOAGhBLXwCcBLABw1qzARBHIDNaAbGNBAG1Q3PgDlocRAGUUtDDACMAOmpYARlgwcQhMAv1sAVCdzSYGJmADmaXCaMAdMABV+GXG6j0eEBbgDcMSxcAEFqW2cosAdPd1wkCDR+exNnN0xA3GwQgFkUfAALdPcAJmjnWIBpGABPAHcsaiI7B2dgsIi0ZyyAEmAANwhqNABfLIBxC1wANWHaCHU+AVHSEV4YCVg2WXkYUtUNLR09A21EWIyPROTWtNd3HsCAGSxrLLMcC0LaGxBRgC65BIdAGMCIAAU1AArGD4c6cCA8ZLkNSaDAzGAjZisRAAVmUAAZlIp-kA)

**Comment by pekkaklarck** [comment]
Tests always needing some content has been a design decision. It would look a bit odd if something like
```
*** Test Cases ***
Example
```
was valid. It's true that with templates you can have a keyword that doesn't require arguments, but that's a rather rare situation and we there's a built-in variable `@{EMPTY}` that can be used explicitly like
```
*** Test Cases ***
Example
    @{EMPTY}
```
if needed.

The above said, I don't feel too strongly about this and we could make a decision that if a test has a template, an empty test means calling the template keyword without arguments. That won't be included in RF 5.1 anymore, but I think I could still make the current error message better even though we already have a release candidate out.

**Comment by pekkaklarck** [comment]
No updates. Closing.
